### PR TITLE
Draw layers that are under sub-displaylists.

### LINKED ISSUE #8325: transform + overflow:scroll + iframe: element not painted
screen shot 2015-11-04 at 06 34 41

```html
<style>

.a {
  transform: translateX(0);
}

.b {
  height: 400px;
  width: 400px;
  background: red;
  overflow: scroll;
}

</style>

<iframe src='data:,x'></iframe>

<div class='a'>
  <div class='b'>
  </div>
</div>
```

### PR DESCRIPTION
**Pull request by notriddle**
Fixes #8325

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8524)

### CODE REVIEW

File: tests/wpt/metadata/MANIFEST.json [hunk 1]
```diff
@@ -29821,6 +29821,24 @@
     "deleted": [],
     "items": {
       "testharness": {
+        "dom/nodes/Node-contains.xml": [
```
Comment (mrobinson) [inline tests/wpt/metadata/MANIFEST.json]: I think the changes in this file are unrelated. It's probably best to revert this particular file and do this in another patch.

**Comment by mrobinson** [comment]
Looks good to me, except for a nit. r=me with that fixed.

**Comment by jdm** [comment]
@notriddle Squash please :)

**Comment by notriddle** [comment]
Done.

**Comment by jdm** [comment]
Thanks!
